docs: merge wsl docs repos

edibotopic opened the pull request:
This merges the docs from the Ubuntu WSL repo to the Ubuntu Pro for WSL repo.
Ultimately, this is with a view to presenting both sets of content on a single documentation website.

Specific changes include:

* moves doc pages and custom configs from Ubuntu WSL repo to UP4W repo
* reorganises docs to better include and surface content from both repos
* rewrites homepage to make more appropriate for distro and app
* adds a Pro feature notice to relevant pages
* lightly rewrites VSCode guide as more generic starter tutorial (to be developed further with community)
* adds redirects for old Ubuntu WSL docs site URLs
* various fixes and touch-ups

After this PR is merged, the UP4W documentation site will be updated with the new content, as normal.
Next, the source repo for the Ubuntu WSL site will be switched to the UP4W repo.
For a while, both sites will have the same content before eventually the ubuntu-wsl URL will become the default.

UDENG-5640

CarlosNihelton commented on docs/reference/distributions.md:
```diff
@@ -0,0 +1,24 @@
+
+Each of these flavours corresponds to a different application on the Microsoft Store, and once installed, they'll create different distros in your WSL. These are the applications we develop and maintain:
+- [Ubuntu](https://apps.microsoft.com/detail/9PDXGNCFSCZV?hl=en-us&gl=US) ships the latest stable LTS release of Ubuntu. When new LTS versions are released, Ubuntu can be upgraded once the first point release is available.
+- [Ubuntu 18.04.6 LTS](https://apps.microsoft.com/detail/9PNKSF5ZN4SW?hl=en-us&gl=US), [Ubuntu 20.04.6 LTS](https://apps.microsoft.com/detail/9MTTCL66CPXJ?hl=en-us&gl=US), and [Ubuntu 22.04.3 LTS](https://apps.microsoft.com/detail/9PN20MSR04DW?hl=en-us&gl=US) are the LTS versions of Ubuntu and receive updates for five years. Upgrades to future LTS releases will not be proposed.
```
I know that you plan to work on this further, but for now I'd rather removing the point-release version out of the app name to avoid inacuracies. Right now, for example, 22.04.3 is incorrect for Jammy Jellyfish, as the current release version is 22.04.5. So it's safer to just call them `Ubuntu 18.04`, ´Ubuntu 22.04` etc.

edibotopic replied:
Strong agree.

edibotopic replied:
https://github.com/canonical/ubuntu-pro-for-wsl/pull/1044/commits/83479b5da3ac4565b10fa953a5c47f94c45bf1f0

CarlosNihelton commented on docs/howto/install-ubuntu-wsl2.md:
````diff
@@ -0,0 +1,133 @@
+```{warning}
+You should always try to use the latest LTS release of Ubuntu, as it offers the best security, reliability and support when using Ubuntu WSL.
+
+Currently we do not have a recommended store for downloading tar and Appx/MSIX files for Ubuntu distros. 
````
The word "store" here confuses me a little bit as I immediately think of "Microsoft Store". Maybe we could use "place", "location" or "page" instead?

edibotopic replied:
Agreed. It is confusing. Might try "recommend location from which to download...".

edibotopic replied:
https://github.com/canonical/ubuntu-pro-for-wsl/pull/1044/commits/6273654f413231ca860decab76079af5818f0039

CarlosNihelton commented on docs/tutorial/develop-with-ubuntu-wsl.md:
```diff
@@ -0,0 +1,221 @@
+
+## Install Node.js and create a new project
+
+Open your WSL Ubuntu terminal and ensure everything is up to date by typing:
```
```suggestion
Open your Ubuntu terminal and ensure everything is up to date by typing:
```
Or "Ubuntu WSL".
We don't use "WSL Ubuntu" anywhere else as far as I know.

edibotopic replied:
Indeed!

edibotopic replied:
https://github.com/canonical/ubuntu-pro-for-wsl/pull/1044/commits/b801a52f00ab58afa939b2c6439bd2bd87d3f0dd

CarlosNihelton commented on docs/tutorial/develop-with-ubuntu-wsl.md:
````diff
@@ -0,0 +1,221 @@
+Then:
+
+```{code-block} text
+$ sudo apt upgrade
+```
+
+Entering your password and pressing `Y` when prompted.
````
````suggestion
$ sudo apt upgrade -y
```

Entering your password when prompted.
````

Not a required change, but you might want to avoid one more prompt.

edibotopic replied:
Good suggestion.

edibotopic replied:
https://github.com/canonical/ubuntu-pro-for-wsl/pull/1044/commits/04e2929a65dd1d32c7806d40c740d4274ed69cdd

CarlosNihelton commented on docs/tutorial/getting-started.md:
````diff
@@ -1,5 +1,10 @@
+    :end-before: <!-- Include end pro -->
+```
+
 Windows Subsystem for Linux ([WSL](https://ubuntu.com/desktop/wsl)) makes it possible to run Ubuntu on a Windows machine.
````
```suggestion
Windows Subsystem for Linux ([WSL](https://ubuntu.com/desktop/wsl)) makes it easy to run Ubuntu on a Windows machine.
```

Or even "easier than [n]ever before"... :)

Technically speaking that was already possible via solutions like Hyper-V or VirtualBox.

edibotopic replied:
You're right. Well observed.

CarlosNihelton commented on docs/tutorial/getting-started.md:
````diff
@@ -1,5 +1,10 @@
+```
+
 Windows Subsystem for Linux ([WSL](https://ubuntu.com/desktop/wsl)) makes it possible to run Ubuntu on a Windows machine.
 Ubuntu Pro for WSL (UP4W) ensures that each new Ubuntu WSL instance that you create will automatically attach to your [Ubuntu Pro](https://ubuntu.com/pro) subscription.
````
Maybe we should finish this sentence giving one reason why folks would want to attach a WSL instance to a Pro subscription?

edibotopic replied:
Yes. I will work on that.

edibotopic replied:
https://github.com/canonical/ubuntu-pro-for-wsl/pull/1044/commits/f7af505b812856bb2af7f50dc14a421a081fe421

CarlosNihelton commented on docs/howto/cloud-init.md:
```diff
@@ -0,0 +1,212 @@
+
+> That example will create a user named `jdoe` and set it as default via `/etc/wsl.conf`, install the packages
+> `ginac-tools` and `octave` and install `vcpkg` from the git repository, since there is no deb or snap of that
+> application (hence the reason for being included in this tutorial - it requires an unusual setup).
```
```suggestion
> application (hence the reason for being included in this guide - it requires an unusual setup).
```

Honestly this document still looks more like a tutorial than a how-to guide in my opinion.

edibotopic replied:
I see your point. However, to me this cloud-init content is for someone who is already competent with Ubuntu on WSL. They will already have "got started" with a tutorial, where they've learned the main concepts. Someone reading this guide will want to simply get something done with cloud-init and this guide largely consists of actions that when followed lead to a specific outcome. It is something that someone could return to multiple times if they forget how to do it. In general, I think we want to avoid having too many tutorials. We want, for example, a tutorial to safely learn Ubuntu WSL and one to learn UP4W. Someone should then have competence to follow guides, explanations and references on more specific or advanced topics. Lastly, sometimes it's a good idea to label something if that's what we want it to become. In this case, I think we should want this cloud-init content to be a how-to guide; if it isn't, then we/I will have to make it more like one.

CarlosNihelton commented on docs/howto/data-science-and-engineering.md:
```diff
@@ -0,0 +1,92 @@
+
+## GNU octave
+
+> ⓘ GNU Octave is software featuring a [high-level programming language](https://en.wikipedia.org/wiki/High-level_programming_language), primarily intended for [numerical computations](https://en.wikipedia.org/wiki/Numerical_analysis). Octave helps in solving linear and nonlinear problems numerically, and for performing other numerical experiments using a language that is mostly compatible with [ ...
```
```suggestion
> ⓘ GNU Octave is software featuring a [high-level programming language](https://en.wikipedia.org/wiki/High-level_programming_language), primarily intended for [numerical computations](https://en.wikipedia.org/wiki/Numerical_analysis). Octave helps in solving linear and nonlinear problems numerically, and for performing other numerical experiments using a language that is mostly compatible with [MATLAB](https://en.wikipedia.org/wiki/MATLAB). [[GNU Octave](https://octave.org/about.html) ]
```

Different from "GNU/Linux" I don't think "GNU Octave" branding requires a slash. I don't see any usage like that in its "About" page.

edibotopic replied:
https://github.com/canonical/ubuntu-pro-for-wsl/pull/1044/commits/71564fa197e8e72fb134c94532adb3e714e8d349

CarlosNihelton reviewed (commented):
I think you found a very sensible way to tackle this task. I have some ideas for your consideration, but nothing major.

edibotopic commented:
> I think you found a very sensible way to tackle this task. I have some ideas for your consideration, but nothing major.

Thanks a lot @CarlosNihelton -- really useful comments and suggestions as usual.

CarlosNihelton commented on docs/howto/install-ubuntu-wsl2.md:
```diff
@@ -0,0 +1,133 @@
+
+In this guide, we’ve shown you how to install Ubuntu WSL on Windows 10 or 11.
+
+We hope you enjoy working with Ubuntu in WSL. Don’t for to check out [our blog](https://ubuntu.com/blog) for the latest news on all things Ubuntu.
```
```suggestion
We hope you enjoy working with Ubuntu in WSL. Don’t forget to check out [our blog](https://ubuntu.com/blog) for the latest news on all things Ubuntu.
```

A contributor opened an issue about this on ubuntu/WSL repo. :)

edibotopic replied:
https://github.com/canonical/ubuntu-pro-for-wsl/pull/1044/commits/fc2087249a8c9fdbbfd93bbac0addc52d4f52b5a

edibotopic commented:
@CarlosNihelton -- I've addressed all your comments and also:

- Added product page and matrix links to the docs site
- Gave the UP4W tutorial a more descriptive url (/getting-started -> /getting-started-with-up4w) and added a redirect

CarlosNihelton reviewed (approved):
Great job as usual!
